src/lib/Generator.class.ts

```diff
@@ -46,57 +47,59 @@ class Generator extends EventEmitter {
 
     this.engine = getEngine(this.endpoint)
 
-    stage.iterator.on('end', count => {
-      this.iterationsIncoming = count
-        for (const $this of this.$thisList) {
-          this.run($this, this.$thisList.length)
```

Comment:
This caused the duplicate URIs: iterating over `this.$thisList`, calling `this.run()` for each URI. The `run()` function (see below) then adds each URI to the same `this.$thisList` again.